### ISSUE TITLE
Documentation links are broken

### ISSUE THREAD

**Issue by jpryne** [rustwasm/wasm-bindgen#1835]
The links to [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/API/WebGPU) in the [RustWasm documentation](https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WebGpu.html) are (all?) dead.

### Steps to Reproduce

1. Go to https://rustwasm.github.io/wasm-bindgen/api/web_sys/struct.WebGpu.html
2. Click on "MDN Documentation"
3. See error

**Comment by Pauan** [comment]
Thanks for pointing this out! It seems to me that this is expected, since WebGPU is a new API which isn't standardized yet, so I'm not surprised that MDN doesn't have a link to it.

What surprises me is that we're generating bindings for a non-standard API. @alexcrichton Any idea why that's happening?

**Comment by jpryne** [comment]
Not surprising to me either, seeing as it isn't yet ratified. Still, I'm chomping at the bit to make use of it...

**Comment by Pauan** [comment]
> Still, I'm chomping at the bit to make use of it...

You can make your own bindings for it, so that way you don't need to wait for it to be standardized.

**Comment by jpryne** [comment]
Yes, I'm tinkering with it via Rust & Vulkan.

**Comment by alexcrichton** [comment]
Yeah unfortunately we simply manufacture links to MDN without actually validating them. All existing links (or those we've found) follow the same pattern so the links are expected to get filled in over time as APIs are documented.

We could perhaps though have a blacklist in web-sys of APIs that don't have working documentation links and route them elsewhere or say-so inline?

**Comment by Pauan** [comment]
@alexcrichton In this particular case, I don't think the issue is with the documentation, the issue is that we're generating API bindings *at all*. The WebGPU API is super unstable, and we've always had the policy of only shipping stable APIs. So there's something deeper wrong going on, maybe with the WebIDL itself.

**Comment by alexcrichton** [comment]
Ah I see, we didn't vet the WebIDL too closely when we ingested all of it, so existing unstable APIs we pulled in is largely by accident. I'll tag this as a breaking change as well because we'd want to remove support for these APIs

**Comment by jpryne** [comment]
As a sidenote, what's a realistic timeline for WebGPU to go live?

**Comment by derekdreery** [comment]
I've found an incorrect doc link at `web_sys::Location::href`. Mentioning it here so I keep track of it.

**Comment by L0g4n** [comment]
@alexcrichton I just wanted to open an issue because I see broken MDN links on `HtmlInputElement` for example. E.g. check out https://docs.rs/web-sys/0.3.45/web_sys/struct.HtmlInputElement.html#method.report_validity. But I am sure there are more.

**Comment by derekdreery** [comment]
If there were a standard way of dealing with them, then people could submit PRs without needing so much input from the maintainers, which would be welcome I'm sure.

**Comment by alexcrichton** [comment]
Documentation links are [generated here](https://github.com/rustwasm/wasm-bindgen/blob/d6399607bff71bad2250540fe9c027ef466f0e6c/crates/webidl/src/util.rs#L82-L89), unfortunately at this time infrastructure isn't in place to pattern-match special cases, we just assume that the mdn links always work.

**Comment by quarksb** [comment]
webgpu is defult option in chrome 113, which will release in Apr 26, 2023 according [chromestatus](https://chromestatus.com/roadmap)
